tests/simple/memory.wast

```diff
@@ -32,5 +32,95 @@
 (memory.grow (i32.const 1))
 (memory.size)
 #assertStack <i32> #maxMemorySize() : < i32 > -1 : .Stack "memory grow unbounded"
-#assertMemory #maxMemorySize() .MemBound "memory grown unbounded"
+#assertEmptyMemory #maxMemorySize() .MemBound "memory grown unbounded"
+
+;; Instantiate with data
+
+(memory (data (i32.const 0) "WASM" ))
+(memory.size)
+#assertTopStack < i32 > 1 "size of stack"
+#assertMemoryData (0, 87) "text to ascii W"
+#assertMemoryData (1, 65) "text to ascii A"
+#assertMemoryData (2, 83) "text to ascii S"
+#assertMemoryData (3, 77) "text to ascii M"
+#assertEmptyMemory 1 1 "memorys string length"
+
+(memory (data (i32.const 100) "WASM" ))
+#assertMemoryData (100, 87) "text to ascii W"
+#assertMemoryData (101, 65) "text to ascii A"
+#assertMemoryData (102, 83) "text to ascii S"
+#assertMemoryData (103, 77) "text to ascii M"
+#assertEmptyMemory 1 1 "memorys string length"
+
+(memory 0 1)
+(data (i32.const 100) "WASM" )
+#assertMemoryData (100, 87) "text to ascii W"
+#assertMemoryData (101, 65) "text to ascii A"
+#assertMemoryData (102, 83) "text to ascii S"
+#assertMemoryData (103, 77) "text to ascii M"
+#assertEmptyMemory 0 1 "memory data separate inst"
+
+;; Store and load
+
+(memory 1)
+(i32.const 1)
+(i64.const 1)
+(i64.store offset=2)
+#assertMemoryData (3, 1) "store is little endian"
+(i32.const 1)
+(i64.const 257)
+(i64.store8 offset=2)
+#assertMemoryData (3, 1) "store8"
+(i32.const 1)
+(i64.const 65537)
+(i64.store16 offset=2)
+#assertMemoryData (3, 1) "store16"
+(i32.const 1)
+(i64.const #pow(i32) +Int 1)
+(i64.store16 offset=2)
+#assertMemoryData (3, 1) "store32"
+#assertEmptyMemory 1 .MemBound ""
+
+(memory 0)
+(i32.const 0)
+(i32.const 0)
+(i32.store8)
+#assertTrap "store to 0 size memory"
+#assertEmptyMemory 0 .MemBound ""
+
+(memory 1)
+(i32.const 65535)
+(i32.const 1)
+(i32.store8)
+#assertMemoryData (65535, 1) "store to memory edge"
+(i32.const 65535)
+(i32.const 1)
+(i32.store16)
+#assertTrap "store outside of size memory"
+#assertEmptyMemory 1 .MemBound ""
+
+(memory 1)
+(i64.store (i32.const 15) (i64.const #pow(i32) -Int 1))
+(i32.const 15)
+(i32.load8_u)
+#assertTopStack <i32> 255 "load8 unsigned"
+
+
+;; Here's a weird bug I don't understand, in load8
+
+(i32.load8_s (i32.const 0))              ;; Works
+#assertTopStack <i32> 0 "load8 signed"
+(i32.load8_s (i32.const 2))              ;; Works
+#assertTopStack <i32> 0 "load8 signed"
+(i32.const 1)
+(i32.load8_s)              ;; Doesn't work!
+#assertTopStack <i32> 0 "load8 signed"
+(i32.load8_s (i32.const 1))              ;; Doesn't work!
```

Comment:
This doesn't work. For some reason, the interpreter fails to unify `( i32 . load8_s ( i32 . const 1 ) ) ` with the rule on line 758 in `wasm.md`.